ksql-engine/src/main/java/io/confluent/ksql/planner/plan/PlanNode.java

```diff
@@ -52,6 +53,8 @@ public DataSourceType getNodeOutputType() {
 
   public abstract List<PlanNode> getSources();
 
+  public abstract List<SelectExpression> getSelectExpressions();
```

Comment:
Let's do this in a follow up PR, but I think it might be a better abstraction to have a class `ProjectionSchema` and have a node return that instead of having two methods (one for schema and one for selects) because these two concepts really are coupled pretty tightly.

Reply:
https://github.com/confluentinc/ksql/issues/3659